### DISCUSSION TITLE
Rog key /m4 mapping

### DISCUSSION THREAD

**Discussion by ben2224** [seerge/g-helper#1827] [General]
Hi,

Was wondering if you would be able to direct me as to how you were able to remap the the M4 key, I seem to have no luck doing it.

Thanks!

**Comment by seerge** [comment]
You go to ``Extra`` in the app and select appropriate action in corresponding dropdown

**Reply by ben2224** [reply]
Sorry, slight miss understanding. 

I know how to remap it using the app, but I'm wondering how its handled in the source code to make it work. (For me, no software picks it up as a key, and there's no exe I can replace - so something in the registry?)

**Reply by seerge** [reply]
Asus emits own events via USB, you won't be able to pick it up using other software

**Reply by seerge** [reply]
Code that handles hotkeys is here https://github.com/seerge/g-helper/blob/main/app/Input/InputDispatcher.cs

**Reply by ben2224** [reply]
Sorry, I don't mean to be a pain but, I dont know C# so don't have the best of ideas as to whats going on. I've managed to get the HID device with the ID in your code - is it just the case of reading the data from that? but I don't know what data I'm looking for.

**Reply by seerge** [reply]
@ben2224 I'm also not a big expert in c# 

But you can ask chatGPT to translate code to the language of your choice :)

You get device and then listen to it all the time
https://github.com/seerge/g-helper/blob/5de44067bc8dfa40c13d902aaf7df0e52673b7f0/app/Input/KeyboardListener.cs#L6

**Reply by ben2224** [reply]
I don't seem to be getting any data, it just returns `b''` (aka no bytes).
